R/validation.R

```diff
@@ -111,16 +111,16 @@ validate_outcomes_are_numeric <- function(outcomes) {
   check <- check_outcomes_are_numeric(outcomes)
 
   if (!check$ok) {
-    bad_cols <- glue::single_quote(names(check$bad_classes))
-    bad_printable_classes <- map(check$bad_classes, glue_quote_collapse)
-    bad_msg <- glue::glue("{bad_cols}: {bad_printable_classes}")
-    bad_msg <- glue::glue_collapse(bad_msg, sep = "\n")
-
-    glubort(
-      "All outcomes must be numeric, but the following are not:",
-      "\n",
-      "{bad_msg}"
-    )
+    bad_col <- map(names(check$bad_classes), ~ cli::format_inline("{.val {.x}}"))
+    bad_class <- map(check$bad_classes, ~ cli::format_inline("{.cls {.x}}"))
+
+    bad_msg <- glue::glue("{bad_col}: {bad_class}")
```

Comment:
Can you use `cli::format_inline()`? With these conversions, we should be able to remove glue as a direct dependency.

Reply:
`format_inline()` collapses everything down into a single string. As does the rest of cli, so currently that's not possible. hardhat uses glue in some other places and cli depends on glue. So I would wait a little and see if Garbor is up for adding vectorized things to cli or try to remove the direct dependency on glue in a separate PR.